src/com/opc/opcconnectionimpl.cpp

```diff
@@ -266,14 +265,12 @@ void COpcConnectionImpl::OnDataChange(COPCGroup & paGroup, COPCItemDataMap & paC
   while(pos){
     OPCItemData *data = paChanges.GetNextValue(pos);
     if (data) {
-      // Possible unpredictable string behavior in a multi-threaded program
-      std::string itemName = WS2S(data->item()->getName());
-      itemList.push_back(new SOpcItemData(itemName.c_str(), (Variant) data->vDataValue));
+      itemList.push_back(new SOpcItemData(WS2S(data->item()->getName()).c_str(), (Variant) data->vDataValue));
     }
   }
 
   // Possible unpredictable string behavior in a multi-threaded program
-  std::string s_groupName = WS2S(paGroup.getName());
+  const std::string s_groupName = static_cast<const std::string>(WS2S(paGroup.getName()));
```

Comment:
why do you need this cast?

Reply:
Just to make sure the group name really is an immutable constant. Previously I had a string conversion problem, somehow groups and items name get corrupted so the OPC items could not be read.

Reply:
So you need a copy of the mutable value? If that is the case I would make the copy explicit and not hide it via a cast which probably exactly does this in the background.
I think this should work by passing the value to the ctor
const std::string s_groupName(WS2S(paGroup.getName()));

Reply:
while I was thinking about your patch I was pondering what is the task of the WS2S? You are calling it a lot. Would it make sense to store the result of this function in the data structures?

Reply:
@MartinMelikMerkumians I have to retain the name because there is another variable with the same name.
@azoitl It's a helper function from OPCClientToolKit library to convert std::wstring to std::string:
```
    static std::string WS2S(const std::wstring &wstr)
    {
        return std::wstring_convert<std::codecvt_utf8<wchar_t>>().to_bytes(wstr);
    }

    static std::wstring S2WS(const std::string &str)
    {
        return std::wstring_convert<std::codecvt_utf8<wchar_t>>().from_bytes(str);
    }
```